verify/verify.go

```diff
@@ -92,7 +92,7 @@ func (db *DB) VerifySignatures(s *data.Signed, role string) error {
 	// Verify that a threshold of keys signed the data. Since keys can have
 	// multiple key ids, we need to protect against multiple attached
 	// signatures that just differ on the key id.
-	seen := make(map[string]struct{})
+	seenKeys := make(map[string]struct{})
 	valid := 0
 	for _, sig := range s.Signatures {
```

Comment:
While we are fixing this: let's not iterate over the signatures first, but instead over the expected keyids (SHA2-256 is fine for now), then we extract the signature for the keyid and check it.

Reply:
I think that the behavior would be strictly equivalent since the first thing we do in `for _, sig := range s.Signatures {` is `if !roleData.ValidKey(sig.KeyID) { continue }`. However if we were iterating over `roleData.KeyIDs` we won't be able to do lookup in `s.Signatures` which is a slice and note a map. We could create a map over `s.Signatures` but that's more code and more opportunities for bugs.

Reply:
I like the python-tuf approach of iterating over expected keyIDs the best, but let's track it another issue and fix it going fwd.